payments-core/src/main/java/com/stripe/android/payments/core/injection/Stripe3DSAuthenticatorModule.kt

```diff
@@ -0,0 +1,60 @@
+package com.stripe.android.payments.core.injection
+
+import android.content.Context
+import com.stripe.android.PaymentAuthConfig
+import com.stripe.android.PaymentRelayStarter
+import com.stripe.android.model.StripeIntent.NextActionData
+import com.stripe.android.networking.AnalyticsRequestExecutor
+import com.stripe.android.networking.AnalyticsRequestFactory
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.payments.Stripe3ds2CompletionStarter
+import com.stripe.android.payments.core.authentication.IntentAuthenticator
+import com.stripe.android.payments.core.authentication.Stripe3DS2Authenticator
+import com.stripe.android.payments.core.authentication.WebIntentAuthenticator
+import com.stripe.android.stripe3ds2.service.StripeThreeDs2ServiceImpl
+import com.stripe.android.stripe3ds2.transaction.MessageVersionRegistry
+import com.stripe.android.view.AuthActivityStarterHost
+import dagger.Module
+import dagger.Provides
+import dagger.multibindings.IntoMap
+import javax.inject.Named
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Provides [IntentAuthenticator] for [NextActionData.SdkData.Use3DS2].
+ * Requires 3ds2 SDK.
+ */
+@Module
+internal class Stripe3DSAuthenticatorModule {
+    @IntentAuthenticatorMap
+    @Provides
+    @IntoMap
+    @IntentAuthenticatorKey(NextActionData.SdkData.Use3DS2::class)
+    internal fun provideStripe3DSAuthenticator(
+        context: Context,
+        @Named(AuthenticationComponent.ENABLE_LOGGING) enableLogging: Boolean,
+        stripeRepository: StripeRepository,
+        webIntentAuthenticator: WebIntentAuthenticator,
+        paymentRelayStarterFactory: (AuthActivityStarterHost) -> PaymentRelayStarter,
+        analyticsRequestExecutor: AnalyticsRequestExecutor,
+        analyticsRequestFactory: AnalyticsRequestFactory,
+        stripe3ds2CompletionStarterFactory: (AuthActivityStarterHost, Int) -> Stripe3ds2CompletionStarter,
+        @IOContext workContext: CoroutineContext,
+        @UIContext uiContext: CoroutineContext
+    ): IntentAuthenticator {
+        return Stripe3DS2Authenticator(
```

Comment:
`Stripe3DSAuthenticatorModule` provides instantiation to the map. We can declare `@Inject` in `Stripe3DS2Authenticator`'s constructor, then use `@Binds` here too, but we will create other Modules that tries to create an `Authenticator` through reflection(because the main SDK doesn't have direct dependency on it), and returns a `NotSupportedAuthenticator` if it fails. Therefore leaves `Stripe3DSAuthenticatorModule` as a reference for createing 3p `authenticator`s

Reply:
For more details please check [this](https://paper.dropbox.com/doc/diagram-Modularizing-Android-Mobile-SDK--BM1lOu1BZcqPNb3XeFFSwNhdAg-M9mrrTib7RFBVN7VimeGx#:uid=891791521002477904119400&h2=Progressive-milestones) part of the modularization doc

Reply:
Please refer to [this](https://git.corp.stripe.com/stripe-internal/android/blob/ca093dac5abb3b898aa7f34b314715a0c3020b82/terminalsdk/core/src/main/kotlin/com/stripe/stripeterminal/dagger/HandoffClientModule.kt#L34-L56) internal link on a similar implemenation from Terminal SDK - the difference is instead of returning null, we'll return a `NotSupportedIntentAuthenticator`, resulting in a (NextActionData.NewType, NotSupportedIntentAuthenticator) in the map